dashboard/app/models/levels/level_group.rb

```diff
@@ -196,4 +196,17 @@ def self.get_survey_results(script, section)
       end
     end.compact
   end
+
+  # @param {User} current_user - The currently signed in user
+  # @param {User} user - The optional user we're trying to see the attempt of
+  # @param {Level} level - The sublevel we'd like the last attempt for
+  def self.get_sublevel_last_attempt(current_user, user, level)
+    if user && current_user && user != current_user
+      # load other user's solution for teachers viewing their students' solution
+      user.last_attempt(level).try(:level_source).try(:data)
+    elsif current_user
+      # load current_user's previous attempt at this puzzle.
+      current_user.last_attempt(level).try(:level_source).try(:data)
+    end
+  end
```

Comment:
I could simplify this further, to only accept a single user as input, and have the haml file call `LevelGroup.get_sublevel_last_attempt(@user || current_user)`. The reasons I chose to leave it like this:
- More similar to the code we already had = less chance of regression
- Having the logic around which user to use here means we can test it
- Having the logic around which user here, it's is likely easier to grok than doing `@user || current_user`

I'm not sure I feel strongly though, and could be convinced otherwise.

Reply:
I'm cool with the separate arguments.  You could apply that expression within this method implementation to deduplicate a bit though:
```ruby
def self.get_sublevel_last_attempt(current_user, user, level)
  (user || current_user).
    try(:last_attempt, level).
    try(:level_source).
    try(:data)
end
```